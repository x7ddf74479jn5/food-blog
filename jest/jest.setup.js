// Optional: configure or set up a testing framework before each test.
// If you delete this file, remove `setupFilesAfterEnv` from `jest.config.js`

// Used for __tests__/testing-library.js
// Learn more: https://github.com/testing-library/jest-dom
import "@testing-library/jest-dom/extend-expect";

window.matchMedia = jest.fn().mockImplementation((query) => {
  return {
    addListener: jest.fn(),
    matches: false,
    media: query,
    onchange: null,
    removeListener: jest.fn(),
  };
});
