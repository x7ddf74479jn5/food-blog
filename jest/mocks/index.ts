export { mockNextImage } from "./image";
export type { NextRouter } from "./router";
export { mockRouter, RouterContext, withMockedRouter } from "./router";
export { setupWindow } from "./window";
