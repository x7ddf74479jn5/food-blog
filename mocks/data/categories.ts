export const categories = {
  rice: { id: "1", slug: "rice", name: "ご飯物", color: "grey" },
  salad: { id: "2", slug: "salad", name: "サラダ", color: "grey" },
  tips: { id: "3", slug: "tips", name: "料理基礎", color: "grey" },
};
