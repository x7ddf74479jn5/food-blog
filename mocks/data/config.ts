import type { TConfig } from "@/types/models/config";

import { dateCommon } from "./utils";

export const mockConfig: TConfig = {
  ...dateCommon,
  apiHost: "https://food-blog.microcms.io/api/v1/",
  host: "https://food-blog-chi.vercel.app/",
  siteTitle: "Food Blog",
  siteDescription: "料理レシピを紹介するブログ",
  siteKeywords: ["料理"],
  siteImage: { url: "https://food-blog-chi.vercel.app/site_image.jpg", width: 100, height: 100 },
  organization: "Pandashark",
  officialSite: "https://next-portfolio-livid.vercel.app/",
  twitterId: "@pandashark6",
  email: "x7ddf74479jn5@gmail.com",
};
