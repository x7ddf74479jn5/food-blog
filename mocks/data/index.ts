export { mockArticles } from "./articles";
export { mockCategories } from "./categories";
export { mockConfig } from "./config";
export { mockPickup } from "./pickup";
export { mockTags } from "./tags";
export { dateCommon } from "./utils";
export { mockWriters } from "./writers";
