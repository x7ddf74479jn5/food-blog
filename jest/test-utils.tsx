import type Queries from "@testing-library/dom/types/queries";
import type { RenderResult } from "@testing-library/react";
import { render } from "@testing-library/react";
import type { RequestHandler } from "msw";
import { setupServer } from "msw/node";
import { SWRConfig } from "swr";

import { SearchProvider } from "@/contexts/search/SearchContext";
import { ViewportProvider } from "@/contexts/viewport/ViewportContext";

import { withMockLegacyRouter, withMockRouter } from "./mocks";

export const Providers: React.ComponentType<{ children?: React.ReactNode }> = ({ children }) => {
  return withMockRouter(
    withMockLegacyRouter(
      <SWRConfig value={{ dedupingInterval: 0, provider: () => new Map() }}>
        <SearchProvider>
          <ViewportProvider>{children}</ViewportProvider>
        </SearchProvider>
      </SWRConfig>
    )
  );
};

const customRender = (ui: React.ReactElement, options = {}): RenderResult<typeof Queries, HTMLElement> => {
  return render(ui, { wrapper: Providers, ...options });
};

// re-export everything
export * from "./mocks";
export * from "@testing-library/react";

// override render method
export { render as originalRender, customRender as render };

export const reTestCase = {
  anyImage: expect.stringMatching(/^(data:image\/gif)|\.(png|webp|jpeg|jpg|svg)$/),
  anyWord: expect.stringMatching(/\w+/),
};

type ErrorComponentProps = {
  message?: string;
};
export const ErrorComponent: React.FC<ErrorComponentProps> = ({ message }) => {
  throw new Error(message);
};

export const setupMockServer = (...handlers: RequestHandler[]) => {
  const server = setupServer(...handlers);
  beforeAll(() => server.listen());
  afterEach(() => server.resetHandlers());
  afterAll(() => server.close());
  return server;
};
