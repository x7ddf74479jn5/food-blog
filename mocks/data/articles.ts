import type { TArticle } from "@/types";

import { categories } from "./categories";
import { tags } from "./tags";
import { dateCommon } from "./utils";
import { writers } from "./writers";

const articleStock = {
  id: "1",
  title: "基本の一番だしの作り方",
  writer: writers.pandashark,
  description:
    "だし昆布と鰹節（削り節）から取った出汁は、味噌汁、そば、うどん、煮物、鍋料理など、多くの和食に使用できます。",
  image: { url: "/images/5026416_s.jpg", height: 427, width: 640 },
  excerpt:
    "だし昆布と鰹節（削り節）から取った出汁は、味噌汁、そば、うどん、煮物、鍋料理など、多くの和食に使用できます。",
  body: `**分量： 800ml**

だし昆布と鰹節（削り節）から取った出汁は、味噌汁、そば、うどん、煮物、鍋料理など、多くの和食に使用できます。

## 材料
- 水：1000ml
- だし昆布：10g
- かつお削り節：40g

## 作り方
- 固く絞った布巾で、昆布の表面の汚れを軽く拭き取ります。
- 鍋に水と昆布を入れます。
- 弱火で７分ほど熱し、鍋底から小さい泡が出てきたら昆布を取り出します。
- 昆布だしを一旦沸騰させてから、削り節を入れ、弱火で１分煮出します。
- ボウルの上にザルを置き、その上にキッチンペーパーを敷いて出汁を濾します。

---

**二番だしの取り方**
1. 鍋に 500ml の水を入れて沸騰させます。
2. 鍋に一番だしのだしがらを入れ、３分煮出します。
3. 一番だしと同じ方法で濾します。

---

## 保存方法
使わない分は保存容器に移して、冷蔵庫で保存してください。冷蔵保存で 2〜3 日は持ちます。

##  ポイント
だし昆布の表面の白い粉は「うま味」成分なので、取り除かずにそのままお使いください。`,
  category: categories.rice,
  tags: [tags.preparation, tags.misoSoup, tags.udon],
  imageOption: null,
};

type TArticleCollection = {
  [key: string]: TArticle;
};

export const articles: TArticleCollection = {
  stock: { ...dateCommon, ...articleStock },
};
