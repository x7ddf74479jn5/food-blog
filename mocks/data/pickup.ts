import { mockArticles } from "@mocks/data";
import { dateCommon } from "@mocks/data/utils";

export const mockPickup = {
  id: "1",
  name: "旬の食材を使ったおすすめレシピ",
  description: "旬の食材を使ったおすすめレシピ",
  startDate: new Date("2021/10/08 00:00"),
  endDate: new Date("2023/02/08 00:00"),
  articles: Object.values(mockArticles),
  ...dateCommon,
};
