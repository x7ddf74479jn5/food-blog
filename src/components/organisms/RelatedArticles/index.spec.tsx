import { mockArticles } from "@mocks/data";
import { render, screen } from "jest/test-utils";
import renderer from "react-test-renderer";

import { RelatedArticles } from ".";

describe("components/organisms/RelatedArticles", () => {
  const mockArticleList = Object.values(mockArticles);
  describe("記事がある", () => {
    it("snapshot", () => {
      const tree = renderer.create(<RelatedArticles relatedArticles={mockArticleList} />).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it("OK: 初期表示が正しい", () => {
      const { container } = render(<RelatedArticles relatedArticles={mockArticleList} />);

      expect(container).toHaveTextContent("関連レシピ");
      const articles = screen.getAllByRole("article");
      expect(articles.length).toBe(mockArticleList.length);
    });
  });

  describe("記事がない", () => {
    it("snapshot", () => {
      const tree = renderer.create(<RelatedArticles relatedArticles={[]} />).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it("OK: 初期表示が正しい", () => {
      const { container } = render(<RelatedArticles relatedArticles={[]} />);

      expect(container).toHaveTextContent("関連レシピ");
      expect(container).toHaveTextContent("関連するレシピは見つかりませんでした");
      const articles = screen.queryAllByRole("article");
      expect(articles.length).toBe(0);
    });
  });
});
