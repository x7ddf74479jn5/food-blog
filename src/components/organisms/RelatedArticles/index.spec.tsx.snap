// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/organisms/RelatedArticles 記事がある snapshot 1`] = `
<section
  className="container p-2 dark:text-white bg-gray-50 dark:bg-gray-700"
>
  <h2
    className="px-2 pb-2 mb-2 border-b-2"
  >
    関連レシピ
  </h2>
  <ul
    className="grid sm:grid-cols-2 md:grid-cols-2 lg:grid-cols-1 gap-2 sm:gap-x-1 md:gap-x-1 lg:gap-x-1 sm:gap-y-3 md:gap-y-3 lg:gap-y-3 pb-2"
  >
    <li>
      <article
        className="flex flex-row max-h-36 "
      >
        <a
          href="/articles/p97vmuno3jdn"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <div
            className="flex-shrink-0 mt-1 mr-4 min-w-[max-content]"
          >
            <span
              style={
                Object {
                  "background": "none",
                  "border": 0,
                  "boxSizing": "border-box",
                  "display": "inline-block",
                  "height": "initial",
                  "margin": 0,
                  "maxWidth": "100%",
                  "opacity": 1,
                  "overflow": "hidden",
                  "padding": 0,
                  "position": "relative",
                  "width": "initial",
                }
              }
            >
              <span
                style={
                  Object {
                    "background": "none",
                    "border": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": "initial",
                    "margin": 0,
                    "maxWidth": "100%",
                    "opacity": 1,
                    "padding": 0,
                    "width": "initial",
                  }
                }
              >
                <img
                  alt=""
                  aria-hidden={true}
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2748%27%20height=%2748%27/%3e"
                  style={
                    Object {
                      "background": "none",
                      "border": 0,
                      "display": "block",
                      "height": "initial",
                      "margin": 0,
                      "maxWidth": "100%",
                      "opacity": 1,
                      "padding": 0,
                      "width": "initial",
                    }
                  }
                />
              </span>
              <img
                alt="基本の一番だしの作り方"
                data-nimg="intrinsic"
                decoding="async"
                onError={[Function]}
                onLoad={[Function]}
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style={
                  Object {
                    "border": "none",
                    "bottom": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": 0,
                    "left": 0,
                    "margin": "auto",
                    "maxHeight": "100%",
                    "maxWidth": "100%",
                    "minHeight": "100%",
                    "minWidth": "100%",
                    "objectFit": "cover",
                    "objectPosition": undefined,
                    "padding": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": 0,
                  }
                }
              />
              <noscript>
                <img
                  alt="基本の一番だしの作り方"
                  data-nimg="intrinsic"
                  decoding="async"
                  loading="lazy"
                  src="/_next/image?url=%2Fimages%2F5026416_s.jpg&w=96&q=75"
                  srcSet="/_next/image?url=%2Fimages%2F5026416_s.jpg&w=48&q=75 1x, /_next/image?url=%2Fimages%2F5026416_s.jpg&w=96&q=75 2x"
                  style={
                    Object {
                      "border": "none",
                      "bottom": 0,
                      "boxSizing": "border-box",
                      "display": "block",
                      "height": 0,
                      "left": 0,
                      "margin": "auto",
                      "maxHeight": "100%",
                      "maxWidth": "100%",
                      "minHeight": "100%",
                      "minWidth": "100%",
                      "objectFit": "cover",
                      "objectPosition": undefined,
                      "padding": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "width": 0,
                    }
                  }
                />
              </noscript>
            </span>
          </div>
        </a>
        <div
          className="flex flex-col flex-grow"
        >
          <a
            href="/articles/p97vmuno3jdn"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            基本の一番だしの作り方
          </a>
          <div
            className="flex-grow"
          >
            <ul
              className="flex flex-wrap gap-x-2 gap-y-1"
            >
              <li>
                <a
                  href="/articles/tags/preparation"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                >
                  <span
                    className="inline-block text-sm leading-tight text-gray-600 dark:text-gray-300"
                  >
                    #仕込み
                  </span>
                </a>
              </li>
              <li>
                <a
                  href="/articles/tags/miso_soup"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                >
                  <span
                    className="inline-block text-sm leading-tight text-gray-600 dark:text-gray-300"
                  >
                    #味噌汁
                  </span>
                </a>
              </li>
              <li>
                <a
                  href="/articles/tags/udon"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                >
                  <span
                    className="inline-block text-sm leading-tight text-gray-600 dark:text-gray-300"
                  >
                    #うどん
                  </span>
                </a>
              </li>
            </ul>
          </div>
        </div>
      </article>
    </li>
    <li>
      <article
        className="flex flex-row max-h-36 "
      >
        <a
          href="/articles/mm8oec5icb"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <div
            className="flex-shrink-0 mt-1 mr-4 min-w-[max-content]"
          >
            <span
              style={
                Object {
                  "background": "none",
                  "border": 0,
                  "boxSizing": "border-box",
                  "display": "inline-block",
                  "height": "initial",
                  "margin": 0,
                  "maxWidth": "100%",
                  "opacity": 1,
                  "overflow": "hidden",
                  "padding": 0,
                  "position": "relative",
                  "width": "initial",
                }
              }
            >
              <span
                style={
                  Object {
                    "background": "none",
                    "border": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": "initial",
                    "margin": 0,
                    "maxWidth": "100%",
                    "opacity": 1,
                    "padding": 0,
                    "width": "initial",
                  }
                }
              >
                <img
                  alt=""
                  aria-hidden={true}
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2748%27%20height=%2748%27/%3e"
                  style={
                    Object {
                      "background": "none",
                      "border": 0,
                      "display": "block",
                      "height": "initial",
                      "margin": 0,
                      "maxWidth": "100%",
                      "opacity": 1,
                      "padding": 0,
                      "width": "initial",
                    }
                  }
                />
              </span>
              <img
                alt="新玉ねぎとトマトのシンプルサラダの作り方"
                data-nimg="intrinsic"
                decoding="async"
                onError={[Function]}
                onLoad={[Function]}
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style={
                  Object {
                    "border": "none",
                    "bottom": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": 0,
                    "left": 0,
                    "margin": "auto",
                    "maxHeight": "100%",
                    "maxWidth": "100%",
                    "minHeight": "100%",
                    "minWidth": "100%",
                    "objectFit": "cover",
                    "objectPosition": undefined,
                    "padding": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": 0,
                  }
                }
              />
              <noscript>
                <img
                  alt="新玉ねぎとトマトのシンプルサラダの作り方"
                  data-nimg="intrinsic"
                  decoding="async"
                  loading="lazy"
                  src="/_next/image?url=%2F2996666_s.jpg&w=96&q=75"
                  srcSet="/_next/image?url=%2F2996666_s.jpg&w=48&q=75 1x, /_next/image?url=%2F2996666_s.jpg&w=96&q=75 2x"
                  style={
                    Object {
                      "border": "none",
                      "bottom": 0,
                      "boxSizing": "border-box",
                      "display": "block",
                      "height": 0,
                      "left": 0,
                      "margin": "auto",
                      "maxHeight": "100%",
                      "maxWidth": "100%",
                      "minHeight": "100%",
                      "minWidth": "100%",
                      "objectFit": "cover",
                      "objectPosition": undefined,
                      "padding": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "width": 0,
                    }
                  }
                />
              </noscript>
            </span>
          </div>
        </a>
        <div
          className="flex flex-col flex-grow"
        >
          <a
            href="/articles/mm8oec5icb"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            新玉ねぎとトマトのシンプルサラダの作り方
          </a>
          <div
            className="flex-grow"
          >
            <ul
              className="flex flex-wrap gap-x-2 gap-y-1"
            >
              <li>
                <a
                  href="/articles/tags/tomato"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                >
                  <span
                    className="inline-block text-sm leading-tight text-gray-600 dark:text-gray-300"
                  >
                    #トマト
                  </span>
                </a>
              </li>
              <li>
                <a
                  href="/articles/tags/onion"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                >
                  <span
                    className="inline-block text-sm leading-tight text-gray-600 dark:text-gray-300"
                  >
                    #玉ねぎ
                  </span>
                </a>
              </li>
              <li>
                <a
                  href="/articles/tags/ooba"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                >
                  <span
                    className="inline-block text-sm leading-tight text-gray-600 dark:text-gray-300"
                  >
                    #大葉
                  </span>
                </a>
              </li>
            </ul>
          </div>
        </div>
      </article>
    </li>
    <li>
      <article
        className="flex flex-row max-h-36 "
      >
        <a
          href="/articles/3y-w7i75kmd"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <div
            className="flex-shrink-0 mt-1 mr-4 min-w-[max-content]"
          >
            <span
              style={
                Object {
                  "background": "none",
                  "border": 0,
                  "boxSizing": "border-box",
                  "display": "inline-block",
                  "height": "initial",
                  "margin": 0,
                  "maxWidth": "100%",
                  "opacity": 1,
                  "overflow": "hidden",
                  "padding": 0,
                  "position": "relative",
                  "width": "initial",
                }
              }
            >
              <span
                style={
                  Object {
                    "background": "none",
                    "border": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": "initial",
                    "margin": 0,
                    "maxWidth": "100%",
                    "opacity": 1,
                    "padding": 0,
                    "width": "initial",
                  }
                }
              >
                <img
                  alt=""
                  aria-hidden={true}
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2748%27%20height=%2748%27/%3e"
                  style={
                    Object {
                      "background": "none",
                      "border": 0,
                      "display": "block",
                      "height": "initial",
                      "margin": 0,
                      "maxWidth": "100%",
                      "opacity": 1,
                      "padding": 0,
                      "width": "initial",
                    }
                  }
                />
              </span>
              <img
                alt="栄養たっぷり小松菜と油揚げのおひたしの作り方"
                data-nimg="intrinsic"
                decoding="async"
                onError={[Function]}
                onLoad={[Function]}
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style={
                  Object {
                    "border": "none",
                    "bottom": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": 0,
                    "left": 0,
                    "margin": "auto",
                    "maxHeight": "100%",
                    "maxWidth": "100%",
                    "minHeight": "100%",
                    "minWidth": "100%",
                    "objectFit": "cover",
                    "objectPosition": undefined,
                    "padding": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": 0,
                  }
                }
              />
              <noscript>
                <img
                  alt="栄養たっぷり小松菜と油揚げのおひたしの作り方"
                  data-nimg="intrinsic"
                  decoding="async"
                  loading="lazy"
                  src="/_next/image?url=%2F4966941_s.jpg&w=96&q=75"
                  srcSet="/_next/image?url=%2F4966941_s.jpg&w=48&q=75 1x, /_next/image?url=%2F4966941_s.jpg&w=96&q=75 2x"
                  style={
                    Object {
                      "border": "none",
                      "bottom": 0,
                      "boxSizing": "border-box",
                      "display": "block",
                      "height": 0,
                      "left": 0,
                      "margin": "auto",
                      "maxHeight": "100%",
                      "maxWidth": "100%",
                      "minHeight": "100%",
                      "minWidth": "100%",
                      "objectFit": "cover",
                      "objectPosition": undefined,
                      "padding": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "width": 0,
                    }
                  }
                />
              </noscript>
            </span>
          </div>
        </a>
        <div
          className="flex flex-col flex-grow"
        >
          <a
            href="/articles/3y-w7i75kmd"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            栄養たっぷり小松菜と油揚げのおひたしの作り方
          </a>
          <div
            className="flex-grow"
          >
            <ul
              className="flex flex-wrap gap-x-2 gap-y-1"
            >
              <li>
                <a
                  href="/articles/tags/aburaage"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                >
                  <span
                    className="inline-block text-sm leading-tight text-gray-600 dark:text-gray-300"
                  >
                    #油揚げ
                  </span>
                </a>
              </li>
              <li>
                <a
                  href="/articles/tags/komatshna"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                >
                  <span
                    className="inline-block text-sm leading-tight text-gray-600 dark:text-gray-300"
                  >
                    #小松菜
                  </span>
                </a>
              </li>
            </ul>
          </div>
        </div>
      </article>
    </li>
    <li>
      <article
        className="flex flex-row max-h-36 "
      >
        <a
          href="/articles/q_y-lyvhew"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <div
            className="flex-shrink-0 mt-1 mr-4 min-w-[max-content]"
          >
            <span
              style={
                Object {
                  "background": "none",
                  "border": 0,
                  "boxSizing": "border-box",
                  "display": "inline-block",
                  "height": "initial",
                  "margin": 0,
                  "maxWidth": "100%",
                  "opacity": 1,
                  "overflow": "hidden",
                  "padding": 0,
                  "position": "relative",
                  "width": "initial",
                }
              }
            >
              <span
                style={
                  Object {
                    "background": "none",
                    "border": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": "initial",
                    "margin": 0,
                    "maxWidth": "100%",
                    "opacity": 1,
                    "padding": 0,
                    "width": "initial",
                  }
                }
              >
                <img
                  alt=""
                  aria-hidden={true}
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2748%27%20height=%2748%27/%3e"
                  style={
                    Object {
                      "background": "none",
                      "border": 0,
                      "display": "block",
                      "height": "initial",
                      "margin": 0,
                      "maxWidth": "100%",
                      "opacity": 1,
                      "padding": 0,
                      "width": "initial",
                    }
                  }
                />
              </span>
              <img
                alt="熊本県産玉ねぎを使った絶品タコライスの作り方"
                data-nimg="intrinsic"
                decoding="async"
                onError={[Function]}
                onLoad={[Function]}
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style={
                  Object {
                    "border": "none",
                    "bottom": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": 0,
                    "left": 0,
                    "margin": "auto",
                    "maxHeight": "100%",
                    "maxWidth": "100%",
                    "minHeight": "100%",
                    "minWidth": "100%",
                    "objectFit": "cover",
                    "objectPosition": undefined,
                    "padding": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": 0,
                  }
                }
              />
              <noscript>
                <img
                  alt="熊本県産玉ねぎを使った絶品タコライスの作り方"
                  data-nimg="intrinsic"
                  decoding="async"
                  loading="lazy"
                  src="/_next/image?url=%2F3129120_s.jpg&w=96&q=75"
                  srcSet="/_next/image?url=%2F3129120_s.jpg&w=48&q=75 1x, /_next/image?url=%2F3129120_s.jpg&w=96&q=75 2x"
                  style={
                    Object {
                      "border": "none",
                      "bottom": 0,
                      "boxSizing": "border-box",
                      "display": "block",
                      "height": 0,
                      "left": 0,
                      "margin": "auto",
                      "maxHeight": "100%",
                      "maxWidth": "100%",
                      "minHeight": "100%",
                      "minWidth": "100%",
                      "objectFit": "cover",
                      "objectPosition": undefined,
                      "padding": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "width": 0,
                    }
                  }
                />
              </noscript>
            </span>
          </div>
        </a>
        <div
          className="flex flex-col flex-grow"
        >
          <a
            href="/articles/q_y-lyvhew"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            熊本県産玉ねぎを使った絶品タコライスの作り方
          </a>
          <div
            className="flex-grow"
          >
            <ul
              className="flex flex-wrap gap-x-2 gap-y-1"
            >
              <li>
                <a
                  href="/articles/tags/rice"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                >
                  <span
                    className="inline-block text-sm leading-tight text-gray-600 dark:text-gray-300"
                  >
                    #ごはん
                  </span>
                </a>
              </li>
              <li>
                <a
                  href="/articles/tags/minced_meat"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                >
                  <span
                    className="inline-block text-sm leading-tight text-gray-600 dark:text-gray-300"
                  >
                    #挽き肉
                  </span>
                </a>
              </li>
              <li>
                <a
                  href="/articles/tags/lettuce"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                >
                  <span
                    className="inline-block text-sm leading-tight text-gray-600 dark:text-gray-300"
                  >
                    #レタス
                  </span>
                </a>
              </li>
              <li>
                <a
                  href="/articles/tags/onion"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                >
                  <span
                    className="inline-block text-sm leading-tight text-gray-600 dark:text-gray-300"
                  >
                    #玉ねぎ
                  </span>
                </a>
              </li>
              <li>
                <a
                  href="/articles/tags/tomato"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                >
                  <span
                    className="inline-block text-sm leading-tight text-gray-600 dark:text-gray-300"
                  >
                    #トマト
                  </span>
                </a>
              </li>
            </ul>
          </div>
        </div>
      </article>
    </li>
  </ul>
</section>
`;

exports[`components/organisms/RelatedArticles 記事がない snapshot 1`] = `
<section
  className="container p-2 dark:text-white bg-gray-50 dark:bg-gray-700"
>
  <h2
    className="px-2 pb-2 mb-2 border-b-2"
  >
    関連レシピ
  </h2>
  <p
    className="text-center"
  >
    関連するレシピは見つかりませんでした
  </p>
</section>
`;
